### PR TITLE
Chore: Converts SaveDashboardAsForm tests to RTL

### LINKED ISSUE #49435: Convert <code class="notranslate">public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #49435

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx [hunk 1]
```diff
@@ -50,8 +50,8 @@ const renderAndSubmitForm = async (
 
   // @ts-ignore strict null error below
```
Comment (joshhunt) [inline public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx:51]: Can you check if this `@ts-ignore` comment is still required? Seems like it wouldn't be?

**Review by joshhunt — APPROVED** [review]

File: public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx (hunk 1 above)
Comment (JoaoSilvaGrafana) [reply public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx]: Yep not needed will remove 👍  (also on the other 'enzyme -> RTL' branch)

File: public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx [hunk 2]
```diff
@@ -50,8 +50,8 @@ const renderAndSubmitForm = async (
...
   await act(async () => {
-    const button = container.find('button[aria-label="Save dashboard button"]');
-    button.simulate('submit');
+    const button = screen.getByRole('button', { name: 'Save dashboard button' });
+    fireEvent.submit(button);
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.test.tsx:53]: can we get rid of the act async bit and just use await userEvent.click instead?

**Review by ashharrison90 — APPROVED** [review]
looks great 🙌

**Pull request merged**